sklearn/metrics/pairwise.py

```diff
@@ -1716,6 +1722,17 @@ def pairwise_distances(X, Y=None, metric="euclidean", *, n_jobs=None,
         .. versionchanged:: 0.23
            Accepts `pd.NA` and converts it into `np.nan`
 
+    dtype : str, type, list of types, default=None
+        Data type required for X and Y. If None, the dtype will be an
```

Comment:
Usually `dtype` is used to specify the output dtype, not the input dtype. I think we should not set the dtype when ensure_2d is false. Which makes me think that maybe we should instead just introduce a validate_input parameter instead of ensure_2d

Reply:
ohh, I was not aware of this. I wil change based on this.

Reply:
Actually, I had picked up this defn of dtype from [here](https://github.com/scikit-learn/scikit-learn/blob/b4678ca8aa0687575e1e1d2809355aa6c6975b16/sklearn/metrics/pairwise.py#L90-L94).

Reply:
I have made few work arounds now. Please take a look, if this sounds non-intuitive, we can go forward with the idea of `validate_input` param.